src/file.cpp

```diff
@@ -278,6 +278,7 @@ namespace
 	}
 }
 # else
+#  undef _BSD_SOURCE
```

Comment:
According to http://man7.org/linux/man-pages/man7/feature_test_macros.7.html any value will do as long as it's defined. However, it does say that it's deprecated so I added a second two lines for _DEFAULT_SOURCE which is the replacement for _BSD_SOURCE.